File: 3635-103-1161-maximum-level-sum-of-a-binary-tree/3635-103-1161-maximum-level-sum-of-a-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxLevelSum = function(root) {
    let count = 0,maxSum = root.val-1,ans = 0
    queue = [root]

    while(queue.length>0){
        count++
        let len = queue.length,sum = 0
        for(let i=0;i<len;i++){
            node = queue.shift()
            sum+= node.val
            if(node.left) queue.push(node.left)
            if(node.right) queue.push(node.right)
        }
        if(sum>maxSum){
            ans = count
            maxSum = sum
        }
    }
    return ans
};

File: 3635-103-1161-maximum-level-sum-of-a-binary-tree/3635-103-1161-maximum-level-sum-of-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxLevelSum(TreeNode root) {
        int count = 0, ans = 0, maxSum = root.val-1;
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);

        while(!queue.isEmpty()){
            count++;
            int len = queue.size(),sum = 0;
            for(int i=0;i<len;i++){
                TreeNode node = queue.poll();
                sum+= node.val;
                if(node.left!=null) queue.add(node.left);
                if(node.right!=null) queue.add(node.right);
            }
            if(sum>maxSum){
                maxSum = sum;
                ans = count;
            }
        }
        return ans;
    }
}